test_cases/stdlib/test_dataclasses.py

```diff
@@ -0,0 +1,17 @@
+# pyright: reportUnnecessaryTypeIgnoreComment=true
+
+import dataclasses
+from typing import Any, Dict, Tuple
+from typing_extensions import assert_type
+
+
+@dataclasses.dataclass
+class D:
+    ...
+
+
+assert_type(dataclasses.astuple(D()), Tuple[Any, ...])
+assert_type(dataclasses.astuple(D(), tuple_factory=tuple), Tuple[Any, ...])
```

Comment:
I think Eric fixed this recently (Any and Unknown being considered different in assert_type()). Let's upgrade pyright to see if it helps.